os-ui/src/pages/CourseModulePage/DiscussionPage/DiscussionForum.tsx

```diff
@@ -12,9 +13,17 @@ function DiscussionForum({ userInfo }:{userInfo:object}): JSX.Element {
   // save typed question
   const [value, setValue] = useState<string>('');
   const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
-    setValue(event.target.value);
+    if (event.target.value.length > 500) event.preventDefault();
+    else setValue(event.target.value);
   };
 
+  // get course informatiom
+  const courseDescriptionState = useSelector<RootState>((state) => state.courseDescriptionRequest) as { entity: CourseDescriptionType, isLoading: boolean, errorMessage: string };
+  const { entity } = courseDescriptionState;
```

Comment:
The entity can be passed down from a parent component to a child component via props, and the child component is only using the entity property, then there is no need to use useSelector in the child component . Instead, you can simply use the entity prop that is being passed down from the parent.